src/werkzeug/routing.py

```diff
@@ -690,7 +691,7 @@ def __init__(
         string: str,
         defaults: Optional[Any] = None,
         subdomain: Optional[str] = None,
-        methods: Optional[Union[Set[str], str, List[str]]] = None,
+        methods: Optional[Iterable[str]] = None,
```

Comment:
Would a `str` be valid as an `Iterable[str]`?
```suggestion
        methods: Optional[Iterable[str]] = None,
```

Reply:
I ran into this problem when typing what `secret_key` could be in ItsDangerous. Since there's an `if isinstance(methods, str)` later on, Mypy *should* account for that.

Reply:
Thanks, I've now fixed this.